Add Critical Deflection

Tiamanti opened the pull request:
Adds Critical Deflection mechanic as option next to Critical table wound.
Uses existing Hack mechanic.
![image](https://github.com/user-attachments/assets/55582e9e-419e-4e9a-a496-ce3081512aa6)
Also fixes critical roll not being colored in message.
![image](https://github.com/user-attachments/assets/37ffb195-95b3-4f45-9670-4277309a3e01)

moo-man commented on src/model/message/test.js:
```diff
@@ -138,4 +140,32 @@ export class WFRPTestMessageModel extends WarhammerTestMessageModel
 
+    static async onApplyCriticalDeflection(ev)
+    {
+        const controlledTokens = [...canvas.tokens.controlled, ...game.user.targets].map(t => t.actor);
+        const targetIntersection = controlledTokens.filter(t => this.test.targets.includes(t));
+        const hitLoc = this.test.hitloc.result;
```
I have generally tried moving away from using controlled tokens.

The paradigm I go for is user targets, falling back to targets of the test if there are no user targets.

Tiamanti replied:
My idea with including controlled tokens here would be that player could click this if they were the target of Critical Hit instead of having to target their own token but if it's the form I'll go with it.

moo-man replied:
That would be covered by `falling back to targets of the test` right?

Tiamanti replied:
In changed version yes.

Tiamanti replied:
Moved this to attack-test.js because I realised this didn't work for Trait Attacks

Tiamanti commented:
@moo-man I've refactored few things but also added fix to #2594 .

It's a partial fix because the same issue is present in several scripts as AOE3 armour is under `itemTypes."wfrp4e-archives3.armour"` and not `itemTypes.armour` and I think I know what those scripts do but would like a go/no go on those.

moo-man commented:
ItemTags.armour should cover both types of armour

moo-man commented:
Yeah don't see any reason to exclude it, unless it has some different rules I don't remember
